### PR TITLE
Display WordPress Posts Widget: Add new option to open links in a new tab

### PR DESCRIPTION
**Pull request by mikeyarce**
open-dwp-new-tab

### CODE REVIEW

File: modules/widgets/wordpress-post-widget.php [hunk 1]
```diff
@@ -116,11 +116,16 @@ public function widget( $args, $instance ) {
...
 			$post_title = ( $single_post->title ) ? $single_post->title : '( No Title )';
 
-			echo '<h4><a href="' . esc_url( $single_post->URL ) . '">' . esc_html( $post_title ) . '</a></h4>' . "\n";
-			if ( ( $instance['featured_image'] == true ) && ( ! empty ( $single_post->featured_image) ) ) {
-				$featured_image = ( $single_post->featured_image ) ? $single_post->featured_image  : '';
-				echo '<a title="' . esc_attr( $post_title ) . '" href="' . esc_url( $single_post->URL ) . '"><img src="' . $featured_image . '" alt="' . esc_attr( $post_title ) . '"/></a>';
+			if ( $instance['open_in_new_window'] == true ) {
+				echo '<h4><a href="' . esc_url( $single_post->URL ) . '" target="_blank">' . esc_html( $post_title ) . '</a></h4>' . "\n";
 			}
+				else {
```
Comment (jeherve) [inline modules/widgets/wordpress-post-widget.php]: Indenting seems to be a bit off, could you take a look?

File: modules/widgets/wordpress-post-widget.php [hunk 2]
```diff
@@ -116,7 +116,11 @@ public function widget( $args, $instance ) {
 			$single_post = $posts_info->posts[$i];
 			$post_title = ( $single_post->title ) ? $single_post->title : '( No Title )';
 
-			echo '<h4><a href="' . esc_url( $single_post->URL ) . '">' . esc_html( $post_title ) . '</a></h4>' . "\n";
+			if ( $instance['open_in_new_window'] == true ) {
+				echo '<h4><a href="' . esc_url( $single_post->URL ) . '" target="_blank">' . esc_html( $post_title ) . '</a></h4>' . "\n";
+			} else {
+				echo '<h4><a href="' . esc_url( $single_post->URL ) . '">' . esc_html( $post_title ) . '</a></h4>' . "\n";		
+			}
```
Comment (samhotchkiss) [inline modules/widgets/wordpress-post-widget.php]: I get a little squeamish about repeating entire lines just to add one attr.

What if, instead, it were something like:

```php
$target = '';
if ( $instance['open_in_new_window'] == true ) {
     $target = ' target="_blank"';
}
echo '<h4><a href="' . esc_url( $single_post->URL ) . '"' . $target . '>' . esc_html( $post_title ) . '</a></h4>' . "\n";
```

That would keep the next dev who touches this code from having to make changes in 2 places if we were to modify something

File: modules/widgets/wordpress-post-widget.php [hunk 3]
```diff
@@ -151,6 +155,12 @@ public function form( $instance ) {
 			$number_of_posts = 5;
 		}
 
+		if ( isset( $instance[ 'open_in_new_window'] ) ) {
+			$open_in_new_window = $instance[ 'open_in_new_window'];
+		} else {
+			$open_in_new_window = false;
+		}
```
Comment (samhotchkiss) [inline modules/widgets/wordpress-post-widget.php]: @georgestephanis, what's the preferred code style here?

Personally, I would probably do:
```php
$open_in_new_window = false;
if ( isset( $instance[ 'open_in_new_window'] ) ) {
	$open_in_new_window = $instance[ 'open_in_new_window'];
}
```

**Comment by mikeyarce** [comment]
That looks much nicer, @samhotchkiss :)

**Comment by jeherve** [comment]
@mikeyarce Want to make the changes suggested by @samhotchkiss?

**Comment by mikeyarce** [comment]
I fixed the suggestions, and then re-committed - not sure how a bunch of other stuff got added?

**Comment by kraftbj** [comment]
@mikeyarce May need to rebase your branch and push -f to override.

**Comment by mikeyarce** [comment]
ok, this got convoluted - going to cherry-pick the stuff from here and do a new PR :)

**Pull request closed without merging**